osf/models/provider.py

```diff
@@ -18,37 +19,67 @@
 from website.util import api_v2_url
 
 
-class PreprintProvider(ObjectIDMixin, ReviewProviderMixin, DirtyFieldsMixin, BaseModel):
-
-    PUSH_SHARE_TYPE_CHOICES = (('Preprint', 'Preprint'),
-                               ('Thesis', 'Thesis'),)
-    PUSH_SHARE_TYPE_HELP = 'This SHARE type will be used when pushing publications to SHARE'
-
-    REVIEWABLE_RELATION_NAME = 'preprint_services'
+class AbstractProvider(TypedModel, ObjectIDMixin, ReviewProviderMixin, DirtyFieldsMixin, BaseModel):
 
     name = models.CharField(null=False, max_length=128)  # max length on prod: 22
+    advisory_board = models.TextField(default='', blank=True)
     description = models.TextField(default='', blank=True)
     domain = models.URLField(blank=True, default='', max_length=200)
     domain_redirect_enabled = models.BooleanField(default=False)
     external_url = models.URLField(null=True, blank=True, max_length=200)  # max length on prod: 25
     email_contact = models.CharField(null=True, blank=True, max_length=200)  # max length on prod: 23
     email_support = models.CharField(null=True, blank=True, max_length=200)  # max length on prod: 23
-    example = models.CharField(null=True, blank=True, max_length=20)  # max length on prod: 5
-    access_token = EncryptedTextField(null=True, blank=True)
-    advisory_board = models.TextField(default='', blank=True)
     social_twitter = models.CharField(null=True, blank=True, max_length=200)  # max length on prod: 8
     social_facebook = models.CharField(null=True, blank=True, max_length=200)  # max length on prod: 8
     social_instagram = models.CharField(null=True, blank=True, max_length=200)  # max length on prod: 8
     footer_links = models.TextField(default='', blank=True)
+    facebook_app_id = models.BigIntegerField(blank=True, null=True)
+    example = models.CharField(null=True, blank=True, max_length=20)  # max length on prod: 5
+    licenses_acceptable = models.ManyToManyField(NodeLicense, blank=True, related_name='licenses_acceptable')
+    default_license = models.ForeignKey(NodeLicense, related_name='default_license',
+                                        null=True, blank=True, on_delete=models.CASCADE)
+    allow_submissions = models.BooleanField(default=True)
+
+    def __unicode__(self):
+        return '{} with id {}'.format(self.name, self.id)
+
+    @property
+    def all_subjects(self):
+        if self.subjects.exists():
+            return self.subjects.all()
+
+    @property
+    def has_highlighted_subjects(self):
```

Comment:
These properties were modified, but don't have tests. I didn't see any obvious AbstractNode tests, but it isn't a bad idea to get at least these tested.

Reply:
I am not sure, how to go about testing AbstractProvider. Shouldn't these properties be tested by provider types that subclass AbstractProvider?

Reply:
These are somewhat tested in `api_tests`:
https://github.com/CenterForOpenScience/osf.io/blob/develop/api_tests/preprint_providers/views/test_preprint_provider_detail.py#L66-L84
https://github.com/CenterForOpenScience/osf.io/blob/develop/api_tests/preprint_providers/views/test_preprint_provider_subjects_list.py#L338-L356

I agree that this behavior could be more thoroughly tested, but it also isn't particularly fragile.